src/In.ProjectEKA.HipService/DataFlow/DataEntryFactory.cs

```diff
@@ -76,29 +80,29 @@ public DataEntryFactory()
             return Option.Some(new EncryptedEntries(processedEntries.AsEnumerable(), keyMaterial));
         }
 
-        private Entry StoreComponentAndGetLink(Entry componentEntry, string transactionId)
+        private Entry StoreComponentAndGetLink(Entry componentEntry, string transactionId,string careContextReference)
         {
             var linkId = Guid.NewGuid().ToString();
             var token = Guid.NewGuid().ToString();
-            var linkEntry = LinkEntry(linkId, token, transactionId);
+            var linkEntry = LinkEntry(linkId, token, transactionId,careContextReference);
             StoreComponentEntry(linkId, componentEntry, token);
             return linkEntry;
         }
 
-        private static Entry EntryWith(string content, string link)
+        private static Entry EntryWith(string content, string link,string careContextReference)
         {
-            return new Entry(content, FhirMediaType, "MD5", link);
+            return new Entry(content, FhirMediaType, "MD5", link,careContextReference);
         }
 
-        private Entry LinkEntry(string linkId, string token, string transactionId)
+        private Entry LinkEntry(string linkId, string token, string transactionId,string careContextReference)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'transactionId'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5414977)